query/encode.go

```diff
@@ -166,6 +173,8 @@ func reflectValue(values url.Values, val reflect.Value) error {
 				del = ','
 			} else if opts.Contains("space") {
 				del = ' '
+			} else if opts.Contains("key") {
```

Comment:
Historically, the behavior has not been well defined, ([this SO question](http://stackoverflow.com/questions/1746507/authoritative-position-of-duplicate-http-get-query-keys) gives a pretty good overview) so I guess calling the PHP/Rails behavior "erroneous" is a little unfair of me.  There is a [new URL spec](https://url.spec.whatwg.org/) that aims to clarify a lot of the ambiguities in RFC 3986, but I'm not sure if it addresses this specific issue.

So yeah, let's just go with `brackets`.